src/aggregations/pipeline-aggregations/moving-function-aggregation.js

```diff
@@ -0,0 +1,128 @@
+ *             .interval('day')
+ *             .agg(esb.sumAggregation('the_sum', 'lemmings'))
+ *             // Relative path to sibling metric `the_sum`
+ *             .agg(esb.movingFunctionAggregation('the_movfn', 'the_sum'))
+ *     )
+ *     .size(0);
+ *
+ * @example
+ * const reqBody = esb.requestBodySearch()
+ *     .agg(
+ *         esb.dateHistogramAggregation('my_date_histo', 'timestamp')
+ *             .interval('day')
+ *             // Use the document count as it's input
+ *             .agg(esb.movingFunctionAggregation('the_movfn', '_count'))
+ *     )
+ *     .size(0);
+ *
+ * @param {string} name The name which will be used to refer to this aggregation.
+ * @param {string=} bucketsPath The relative path of metric to aggregate over.
+ * @param {string=} window 	The size of window to "slide" across the histogram.
+ * @param {string=} script The script that should be executed on each window of data.
+ *
+ * @extends PipelineAggregationBase
+ */
+class MovingFunctionAggregation extends PipelineAggregationBase {
+    // eslint-disable-next-line require-jsdoc
+    constructor(name, bucketsPath, window, script) {
+        super(name, 'moving_fn', ES_REF_URL, bucketsPath);
+
+        if (!isNil(window)) this._aggsDef.window = window;
+        if (!isNil(script)) this._aggsDef.script = script;
+    }
+
+    /**
+     * @override
+     * @throws {Error} This method cannot be called on MovingFunctionAggregation
+     */
+    format() {
+        console.log(`Please refer ${ES_REF_URL}`);
+        throw new Error('format is not supported in MovingFunctionAggregation');
+    }
+
+    /**
+     * Sets the size of window to "slide" across the histogram. Optional.
+     *
+     * @example
+     * const agg = esb.movingFunctionAggregation('the_movfn', 'the_sum')
+     *     .window(30)
+     *
+     * @param {number} window Default is 5
+     * @returns {MovingFunctionAggregation} returns `this` so that calls can be chained
+     */
+    window(window) {
+        this._aggsDef.window = window;
+        return this;
+    }
+
+    /**
+     * Sets shift of window position. Optional.
+     *
+     * @example
+     * const agg = esb.movingFunctionAggregation('the_movfn', 'the_sum')
+     *     .shift(30)
+     *
+     * @param {number} shift Default is 0
+     * @returns {MovingFunctionAggregation} returns `this` so that calls can be chained
+     */
+    shift(shift) {
+        this._aggsDef.shift = shift;
+        return this;
+    }
+
+    /**
+     * Sets the script that should be executed on each window of data. Required.
+     *
+     * @example
+     * const agg = esb.movingFunctionAggregation('the_movfn', 'the_sum', "MovingFunctions.unweightedAvg(values)"")
+     *     .script("MovingFunctions.unweightedAvg(values)")
+     *
+     * @param {string} script
+     * @returns {MovingFunctionAggregation} returns `this` so that calls can be chained
+     */
+    script(script) {
+        if (isNil(script)) invalidModelParam(script);
```

Comment:
This validation is not correct. Probably deleting this line should be okay since custom user scripts are also allowed.

Reply:
Nice catch! Thanks for the good work.